src/loaders/IVolumeLoader.ts

```diff
@@ -18,13 +19,79 @@ export class LoadSpec {
   }
 }
 
+export type LoadSpecExtent = {
+  minx: number;
+  miny: number;
+  minz: number;
+  maxx: number;
+  maxy: number;
+  maxz: number;
+};
+
+/** Converts `LoadSpec` sub-region fields (`minx`, `maxy`, etc.) from [0, 1) range to pixels */
+export function convertLoadSpecRegionToPixels<T extends Partial<LoadSpecExtent>>(
+  loadSpec: T,
+  sizex: number,
+  sizey: number,
+  sizez: number
+): T & LoadSpecExtent {
+  const minx = loadSpec.minx !== undefined ? Math.floor(sizex * loadSpec.minx) : 0;
+  let maxx = loadSpec.maxx !== undefined ? Math.ceil(sizex * loadSpec.maxx) : sizex;
+  const miny = loadSpec.miny !== undefined ? Math.floor(sizey * loadSpec.miny) : 0;
+  let maxy = loadSpec.maxy !== undefined ? Math.ceil(sizey * loadSpec.maxy) : sizey;
+  const minz = loadSpec.minz !== undefined ? Math.floor(sizez * loadSpec.minz) : 0;
+  let maxz = loadSpec.maxz !== undefined ? Math.ceil(sizez * loadSpec.maxz) : sizez;
+
+  // ensure it's always valid to specify the same number at both ends and get a single slice
+  if (minx === maxx && minx < sizex) {
+    maxx += 1;
+  }
+  if (miny === maxy && miny < sizey) {
+    maxy += 1;
+  }
+  if (minz === maxz && minz < sizez) {
+    maxz += 1;
+  }
+
+  return { ...loadSpec, minx, maxx, miny, maxy, minz, maxz };
+}
```

Comment:
a unit test would be good to test this function against edge cases to ensure it always does what is intended